### PR TITLE
Video eagerload

### PR DESCRIPTION
**Pull request by philipqnguyen**
These changes add:
- `$content_owner.videos.includes(:claim)` will so eager-load assets. Allowing you to do this: `$content_owner.videos.includes(:claim).first.claim.asset.id`.
- Add `Claim#source` that returns the source of the claim.

### CODE REVIEW

File: spec/requests/as_content_owner/content_owner_spec.rb [hunk 1]
```diff
@@ -53,19 +53,15 @@
     describe '.includes(:claim)' do
       let(:videos) { $content_owner.videos.includes(:claim) }
       let(:video_with_claim) { videos.find{|v| v.claim.present?} }
+      let(:video_with_claim_with_asset) do
+        videos.find{|v| v.try(:claim).try(:asset)}
+      end
```
Comment (claudiofullscreen) [inline spec/requests/as_content_owner/content_owner_spec.rb]: I think we can get rid of this, and just use the previous `video_with_claim`. 

In other words, every video with claim will also have an asset, and we just expect it to be eager-loaded too.

File: spec/requests/as_content_owner/content_owner_spec.rb [hunk 2]
```diff
@@ -53,19 +53,15 @@
...
 
       specify 'eager-loads the claim of each video' do
```
Comment (claudiofullscreen) [inline spec/requests/as_content_owner/content_owner_spec.rb]: and change this to

``` ruby
specify 'eager-loads the claim of each video and its asset' do
  expect(video_with_claim.claim).to be_a Yt::Claim
  expect(video_with_claim.claim.id).to be_a String
  expect(video_with_claim.claim.video_id).to eq video_with_claim.id

  expect(video_with_claim.claim.asset).to be_a Yt::Asset
  expect(video_with_claim.claim.asset.id).to be_a String 
end
```

File: lib/yt/models/claim.rb [hunk 3]
```diff
@@ -114,13 +114,12 @@ def audiovisual?
...
-      # @return String of the claim's source.
+      # @return [Yt::Models::Claim] the source of the claim
       def source
-        origin = @data.fetch 'origin', nil
-        origin.fetch 'source', nil if origin
+        @data.fetch('origin', {})['source']
       end
 
-      # @return the first asset of the claim.
+      # @return [Yt::Models::Claim] the asset of the claim
```
Comment (claudiofullscreen) [inline lib/yt/models/claim.rb]: Is this `Yt::Models::Claim` or `[Yt::Models::Asset]`?

File: lib/yt/models/claim.rb [hunk 4]
```diff
@@ -114,13 +114,12 @@ def audiovisual?
         content_type == 'audiovisual'
       end
 
-      # @return String of the claim's source.
+      # @return [Yt::Models::Claim] the source of the claim
```
Comment (claudiofullscreen) [inline lib/yt/models/claim.rb]: I think the source is actually a string, such as `webUpload`.

File: lib/yt/models/claim.rb (hunk 3 above)
Comment (philipqnguyen) [inline lib/yt/models/claim.rb]: Yes, it would be asset. Fixing now =D

File: lib/yt/models/claim.rb (hunk 4 above)
Comment (philipqnguyen) [inline lib/yt/models/claim.rb]: Yes it is

**Pull request merged**